lib/modules/manager/woodpecker/extract.ts

```diff
@@ -5,6 +5,24 @@ import { getDep } from '../dockerfile/extract';
 import type { ExtractConfig, PackageFileContent } from '../types';
 import type { WoodpeckerConfig } from './types';
 
+function woodpeckerVersionDecider(
+  woodpeckerConfig: WoodpeckerConfig
+): Array<keyof WoodpeckerConfig> {
+  const keys: Array<keyof WoodpeckerConfig> = [];
+
+  if ('clone' in woodpeckerConfig) {
+    keys.push('clone');
+  }
+
+  if ('steps' in woodpeckerConfig) {
+    keys.push('steps');
+  } else if ('pipeline' in woodpeckerConfig) {
+    keys.push('pipeline');
+  }
+
+  return keys;
```

Comment:
```suggestion
  const keys = ['clone', 'steps', 'pipeline'];
  return Object.keys(woodpeckerConfig).filter(key => keys.includes(key)) as (keyof WoodpeckerConfig)[];
```

This would preserve the dependency order which is important.
Otherwise it can cause hidden side effects when the same dependency occures multiple times.